worlds/hk/__init__.py

```diff
@@ -165,75 +168,138 @@ def white_palace_exclusions(self):
-                        self.create_location(location_name)
+                        loc1.place_locked_item(item1)
+                        loc2.place_locked_item(item2)
+                elif item_name == "Shade_Cloak" and self.world.SplitMothwingCloak[self.player]:
+                    loc = self.create_location(location_name)
+                    if self.world.random.randint(0, 1):
+                        item = self.create_item("Left_Mothwing_Cloak")
+                    else:
+                        item = self.create_item("Right_Mothwing_Cloak")
+                    item.classification = ItemClassification.progression
+                    if randomized:
                         pool.append(item)
-            # elif option_key not in logicless_options:
-            else:
-                for item_name, location_name in zip(option.items, option.locations):
-                    if location_name in wp_exclusions and location_name != 'King_Fragment':
-                        continue
+                    else:
+                        loc.place_locked_item(item)
+                else:
                     item = self.create_item(item_name)
+                # self.create_location(location_name).place_locked_item(item)
                     if location_name == "Start":
-                        self.world.push_precollected(item)
+                        if (item_name == "Focus" and self.world.RandomizeFocus[self.player]) or (item_name == "Swim"
+                                and self.world.RandomizeSwim[self.player]) or (item_name in ["Upslash", "Leftslash",
+                                "Rightslash"] and self.world.RandomizeNail[self.player]):
+                            pool.append(item)
+                        else:
+                            self.world.push_precollected(item)
                     else:
-                        self.create_location(location_name).place_locked_item(item)
-        for i in range(self.world.EggShopSlots[self.player].value):
-            self.create_location("Egg_Shop")
-            pool.append(self.create_item("Geo_Rock-Default"))
+
+                        location = self.create_location(location_name)
+                        if not vanilla and location_name in wp_exclusions:
+                            if location_name == 'King_Fragment':
+                                excluded = True
+                            else:
+                                vanilla = True
+                        if excluded:
+                            location.progress_type = LocationProgressType.EXCLUDED
+                        if vanilla:
+                            location.place_locked_item(item)
+                        else:
+                            pool.append(item)
+
+                if location_name in wp_exclusions:
+                    logging.debug(f"World {self.world.player_name[self.player]} - Item {item_name} - Location {location_name} - vanilla={vanilla!r}, excluded={excluded!r}")
```

Comment:
This was my own accidentally left-in debug code, it can be deleted outright.

Reply:
Done